src/main/io/osd.c

```diff
@@ -395,7 +395,10 @@ static void osdDrawSingleElement(uint8_t item)
     case OSD_HOME_DIR:
         if (STATE(GPS_FIX) && STATE(GPS_FIX_HOME)) {
             if (GPS_distanceToHome > 0) {
-                const int h = GPS_directionToHome - DECIDEGREES_TO_DEGREES(attitude.values.yaw);
+				int h = GPS_directionToHome - DECIDEGREES_TO_DEGREES(attitude.values.yaw);
+				// If MAG_Heading is not available, yaw starts with zero in a random direction. With GPS course over ground this can be roughly corrected
+				if (!FLIGHT_MODE(MAG_MODE)) h = GPS_directionToHome - DECIDEGREES_TO_DEGREES(gpsSol.groundCourse);
```

Comment:
What is the logic of using `FLIGHT_MODE(MAG_MODE)` here?

Are you trying to check if a compass is available? If so you should use `sensors(SENSOR_MAG)` instead.

Reply:
Sorry for introducing all that newbie and whitespace errors. My editor (notepad++) introduces them. I'll try to revert them...
I tried to check if compass is actually not used. Could this being check with !sensors(SENSOR_MAG) ?

Reply:
Yes. `sensors()` checks if a given sensor is initialised. `FLIGHT_MODE()` tests if a given flight mode is activated.

In fact this check should be `sensors(SENSOR_GPS)` (i.e. "do I have GPS?" rather than "do I not have a compass?").